### ISSUE TITLE
Logging against Elastic Search with X-Pack Security Enabled

### ISSUE THREAD

**Issue by dandrejvv** [ReactiveMarkets/NLog.Targets.ElasticSearch#50]
It seems as that this plugin doesn't cater for Elastic Search when X-Pack is installed with the Security feature switched on.
It seems to use OAuth security. Would it be possible to allow this NLog plugin to allow one to configure it to authenticate against Elastic Search's security feature in order to log to Elastic?
Or is there some other feature in Elastic that I'm missing?
At the moment I am logging to Elastic without authentication.

**Comment by markmcdowell** [comment]
I haven't tried it but https://www.elastic.co/guide/en/x-pack/6.0/http-clients.html seems to imply it's just basic authentication. 

Have you tried using the requireAuth setting?

**Comment by dandrejvv** [comment]
Hi @markmcdowell. Thanks for the help. I see now how it works. I haven't noticed those settings in the plugin. I'll try it out and let you know.

**Comment by dandrejvv** [comment]
Ok, so I've tested it and it is working.
When I specify the `requireAuth` to be `true` and the `username` with the `password` it logs the message on Elastic with credentials.
If I omit those, it fails but in the logs (NLog's internal logging) it doesn't say why it failed:

    2017-12-08 08:18:09.8820 Error Failed to send log messages to elasticsearch: status=401, message="One or more errors occurred."
    2017-12-08 08:18:09.8881 Trace Failed to send log messages to elasticsearch: result=Unsuccessful low level call on POST: /_bulk
    2017-12-08 08:18:09.8881 Error Error while sending log messages to elasticsearch: message="One or more errors occurred."

**Comment by markmcdowell** [comment]
Ok cool, the status=401 is unauthorised 
		
		

		Thanks, Mark
	




On Fri, Dec 8, 2017 at 6:21 AM +0000, "Dandré" <notifications@github.com> wrote:












Ok, so I've tested it and it is working.

When I specify the requireAuth to be true and the username with the password it logs the message on Elastic with credentials.

If I omit those, it fails but in the logs (NLog's internal logging) it doesn't say why it failed:
2017-12-08 08:18:09.8820 Error Failed to send log messages to elasticsearch: status=401, message="One or more errors occurred."
2017-12-08 08:18:09.8881 Trace Failed to send log messages to elasticsearch: result=Unsuccessful low level call on POST: /_bulk
2017-12-08 08:18:09.8881 Error Error while sending log messages to elasticsearch: message="One or more errors occurred."

**Comment by kcm** [comment]
Hey, we tried getting this working with Logstash using HTTPS, but without auth. We're using a self-signed certificate on the LS side, with the CA root added to the NLog-side MMC cert store. No dice. We then added basic auth and enabled `requireAuth`, with no change or progress. Logging shows no real errors.

Is `requireAuth` + auth required to support an HTTPS protocol URI? It didn't work either way, but it would be nice to know if HTTPS would work without auth going forward.
